packages/components/modal/src/components/osds-modal/osds-modal.tsx

```diff
@@ -36,31 +34,35 @@ export class OsdsModal implements OdsModalAttribute, OdsModalMethod {
    */
   @Method()
   async close(): Promise<void> {
-    this.controller.close();
+    this.masked = true;
   }
 
   /**
    * @see OdsModalMethods.open
    */
   @Method()
   async open(): Promise<void> {
-    this.controller.open();
+    this.masked = false;
   }
 
-  componentWillRender(): void {
-    document.body.appendChild(this.el);
+  @Watch('masked')
+  watchOpenedStateHandler(masked: boolean) {
     const directChildren = Array.from(document.body.children);
-    for (const child of directChildren) {
-      if (child !== this.el && child.nodeName !== 'SCRIPT') {
-        if (!this.masked) {
-          child.setAttribute('inert', '');
-        } else {
-          child.removeAttribute('inert');
-        }
+    const filteredChildren = directChildren.filter(child => child !== this.el && child.nodeName !== 'SCRIPT');
```

Comment:
not sure it's actually better as you go through 2 loops instead of one

Reply:
It felt like a separating concerns to me. But I can switch this part of the code back to its previous state